pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -241,28 +243,30 @@ func convertEndpoint(service *model.Service, servicePort *networking.Port,
 		instancePort = servicePort.Number
 	}
 
-	tlsMode := getTLSModeFromWorkloadEntry(endpoint)
+	tlsMode := getTLSModeFromWorkloadEntry(wle)
 	sa := ""
-	if endpoint.ServiceAccount != "" {
-		sa = spiffe.MustGenSpiffeURI(service.Attributes.Namespace, endpoint.ServiceAccount)
+	if wle.ServiceAccount != "" {
+		sa = spiffe.MustGenSpiffeURI(service.Attributes.Namespace, wle.ServiceAccount)
 	}
+	labels := labelutil.AugmentLabels(wle.Labels, clusterID, wle.Locality, network.ID(wle.Network))
```

Comment:
We probably need a lot of the same logic that we embed into the `serviceregistry/kube/controller/` to determine which network the endpoint is on:
* cidr range (sort of legacy, part of meshNetworks, but still valid)
* topology label. in this case the `wle.Network` is probably more valid, but we might want to support both
* default network for cluster, if not set on WLE (determined from `topology.istio.io/network` label on the system namespace; see https://github.com/istio/istio/pull/28126)

This would probably require a bit of restructuring to allow re-using this. And now we can run `serviceentry` registries, so we have to support this per-cluster for each registry type. Probably best as a follow-up task since this is already brand new functionality.

Reply:
>cidr range (sort of legacy, part of meshNetworks, but still valid)

Good point

>topology label. in this case the wle.Network is probably more valid, but we might want to support both

This is already supported, wle.network override that one from wle.labels.

>default network for cluster, if not set on WLE (determined from topology.istio.io/network label on the system namespace; see label system namespace with default network for cluster #28126)

It implicitly means wle network should be same as the cluster.  I am not sure if this is right.

I think the order should be: wle.Network > wle.Labels >  cluster namespace label. > cidr(maybe should not support it as it has been deprecated)